### DISCUSSION TITLE
Workaround for "Missing environment variable in config" validation

### DISCUSSION THREAD

**Discussion by GreyTeardrop** [vectordotdev/vector#21553] [Q&A] [answered]
Hi!

I'm looking for a way to workaround or suppress Vector's "Missing environment variable in config" validation. The reason is I'm trying to attach the schema generated by `vector generate-schema` to Vector's YAML. There are two ways IDEs support doing that:
```yaml
# $schema: ./vector.schema.json
```
and
```yaml
# yaml-language-server: $schema=./vector.schema.json
```

Unfortunately, both options include `$schema` in a comment so Vector validation fails with
```
x Missing environment variable in config. name = "schema"
```

**Comment by jszwedko** [comment, marked as answer]
Ah, interesting. You could just `export schema=""` before running `vector validate`. Vector will end up ignoring the line as a comment anyway. I can't think of another good workaround 🤔

**Reply by GreyTeardrop** [reply]
Thank you, I'll use that workaround for now 👍 

As Vector would eventually ignore YAML comments, is it possible long-term not to validate env var references there?

**Reply by jszwedko** [reply]
Yeah, that is being tracked by https://github.com/vectordotdev/vector/issues/797

**Reply by GreyTeardrop** [reply]
Awesome, thank you!
